src/scripts/browserDeviceProfile.js

```diff
@@ -1141,6 +1141,24 @@ export default function (options) {
         Conditions: h264CodecProfileConditions
     });
 
+    if (browser.web0s && supportsDolbyVision(options)) {
+        // Disallow direct playing of DOVI media in containers not mp4.
+        // This paired with the "Prefer fMP4-HLS Container" client playback setting enables DOVI playback on webOS.
+        profile.CodecProfiles.push({
+            Type: 'Video',
+            Container: '-mp4',
+            Codec: 'hevc',
+            Conditions: [
+                {
+                    Condition: 'EqualsAny',
+                    Property: 'VideoRangeType',
+                    Value: 'SDR|HDR10|HLG',
```

Comment:
In theory, this should work too (for DoVi without fallback). It would be nice if you could check (_I only did a rough check_).
```suggestion
                    Condition: 'NotEquals',
                    Property: 'VideoRangeType',
                    Value: 'DOVI',
```
With `NotEquals`, `Value` cannot be a list - we need a new `NotEqualsAny` condition.

FYI, there is an `ApplyConditions` field to activate conditions. I think we should read Dolby Vision with HDR10 fallback as `DOVI,HDR10` when checking the conditions and check both (_requires server-side changes_). So if we need to disable any fallback (_user settings_), we can add (_not tested_):
```js
{
    ApplyConditions: [
        {
            Condition: 'EqualsAny',
            Property: 'VideoRangeType',
            Value: 'DOVI', // with server-side changes, 'DOVI,HDR10' will match
            IsRequired: false
        }
    ],
    Conditions: [
        {
            Condition: 'Equals',
            Property: 'VideoRangeType',
            Value: '', // disable any non-mp4 video with DoVi
            IsRequired: false
        }
    ]
}
```

Reply:
Setting "NotEquals" to DOVI indeed works to kickoff transmux for DOVI content, identical to how the current lines behave. With this and "Enable fMP4-HLS", I am able to view DV Profile 5 media.

It appears the server-side is aligning to implement this fix by introducing the new VideoRangeTypes "DOVIWithHDR10" and "DOVIWithHLG" for fallback-able DV. With that, I thought I'd just update lines 930 and 936 to include them by default for all HDR/HLG supporting clients, and leave this block as-is.

But looking into ApplyConditions, I don't understand how it's to be used. For instance, doing this works (both P8 and P5 remuxed and playback in DOVI, and HDR direct played as expected -- with the latest patches from the server's PR in place)
 ``` 
    if (browser.web0s && supportsDolbyVision(options)) {
        // Disallow direct playing of DOVI media in containers not mp4.
        // This paired with the "Prefer fMP4-HLS Container" client playback setting enables DOVI playback on webOS.
        profile.CodecProfiles.push({

            ApplyConditions: [
                {
                    Condition: 'EqualsAny',
                    Property: 'VideoRangeType',
                    Value: 'DOVI|DOVIWithHDR10|DOVIWithHLG', // with server-side changes, 'DOVI,HDR10' will match
                    IsRequired: false
                }
            ],
            Conditions: [
                {
                    Condition: 'Equals',
                    Property: 'VideoRangeType',
                    Value: '', // disable any non-mp4 video with DoVi
                    IsRequired: false
                }
            ]
        });
    }
```
But I don't fully understand how this is working. Can you please show me some more examples of it's usage, and let me know if what I'm doing here is the same as what you wrote about?

Reply:
> It appears the server-side is aligning to implement this fix by introducing the new VideoRangeTypes "DOVIWithHDR10" and "DOVIWithHLG" for fallback-able DV.

I've been thinking about how we can get around the introduction of a new video range.

If the server will support combined video ranges, we can replace `DOVIWithHDR10` with `[VideoRangeType.DOVI, VideoRangeType.HDR10]`. This means that  the video has a Dolby Vision stream with HDR10 fallback.

Having `HDR10` in the `Condition` will allow playing HDR10 fallback at least.
But if we want to only play Dolby Vision, we can disallow HDR10 if there is also DOVI.

To apply the condition if there is DOVI, we can add the `ApplyCondition`:
```
ApplyConditions: [
    {
        Condition: 'EqualsAny',
        Property: 'VideoRangeType',
        Value: 'DOVI',
        IsRequired: false
    }
]
```
Then, to block any video range:
```
Conditions: [
    {
        Condition: 'Equals',
        Property: 'VideoRangeType',
        Value: '', // not sure if this will work
        IsRequired: false
    }
]
```

> But I don't fully understand how this is working. Can you please show me some more examples of it's usage, and let me know if what I'm doing here is the same as what you wrote about?

I haven't found any example. I only have the idea how it works (https://github.com/jellyfin/jellyfin/blob/231daa38dda8a572d41693fa458f74940ca53949/MediaBrowser.Model/Dlna/StreamBuilder.cs#L1583):

- First, the server checks the "apply" conditions.
- If the video passes them, the server checks the conditions.

In theory, your code (with `DOVIWithHDR10` and `DOVIWithHLG`) can be used to proof the idea.
You need these properties in the condition (I just omitted them):
```
profile.CodecProfiles.push({
    Type: 'Video',
    Container: '-mp4',
    Codec: 'hevc',
...
```
Otherwise, the condition applies to any media (audio, video) and any codec.

Also, I think this condition is in addition to the condition from this PR and should be controlled by the user through the playback settings.

Reply:
Ah, need to target Video. I was also thrown off by not needing a -mp4 with this approach. I can confirm the setup does work with below-
```
     if (browser.web0s && supportsDolbyVision(options)) {
        profile.CodecProfiles.push({
            Type: 'Video',
            Container: '-mp4', //even without this line it works
            Codec: 'hevc',
            ApplyConditions: [
                {
                    Condition: 'EqualsAny',
                    Property: 'VideoRangeType',
                    Value: 'DOVI|DOVIWithHDR10|DOVIWithHLG', 
                    IsRequired: false
                }
            ],
            Conditions: [
                {
                    Condition: 'Equals',
                    Property: 'VideoRangeType',
                    Value: '', // disable whatever ended up matching in ApplyConditions from direct play, right? this will not apply if none of ApplyCondition match
                    IsRequired: false
                }
            ]
        });
    }
```
--
I remember taking a look at server to see if multiple videorangetypes could be supplied, but I found it too complex to change -- I remember too much of the code relied on there being one videorangetype. I felt the changes were not worth the undertaking to allow this feature to work

As a WebOS user, I don't think I would ever want to change the settings away from how it works right now - perhaps only making the "Enable fMP4" option a default for WebOS. If we're okay to take that up in a separate PR, do you recommend the above approach over the current implementation for this PR?

Reply:
> even without this line it works

Without `Container: '-mp4'` Jellyfin won't direct play MP4+DOVI (will always remux), no?

> As a WebOS user, I don't think I would ever want to change the settings away from how it works right now - perhaps only making the "Enable fMP4" option a default for WebOS.

Some users may only want direct playback, even if it's an HDR10/HLG fallback - they just don't want remux.

> If we're okay to take that up in a separate PR, do you recommend the above approach over the current implementation for this PR?

Current implementation is fine. Other options require changes in the server.